src/condor_ce_tools.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python
+
+import errno
+import os
+import tempfile
+import textwrap
+import time
+from subprocess import Popen, PIPE, STDOUT
+
+# Excluding submit file so the respective scripts
+# can generate it as they see fit
+JOB_FILES = ['stdout', 'stderr', 'log']
+
+class CondorRunException(Exception):
+    pass
+
+class CondorUserException(Exception):
+    pass
+
+def print_formatted_msg(msg):
+    print "*"*80
+    for line in textwrap.wrap(msg, 80):
+        print line
+    print "*"*80
+
+def print_timestamped_msg(msg):
+    print_formatted_msg("%s %s" % (time.strftime('%F %T'), msg))
+
+def run_command(command):
+    p = Popen(command, stdout=PIPE, stderr=STDOUT)
+    stdout, stderr = p.communicate()
+    return p.returncode, stdout, stderr
```

Comment:
stderr is always going to be blank here because you are redirecting stderr to stdout above. If you want to separate the two streams, you should do
    p = Popen(command, stdout=PIPE, stderr=PIPE)
If not, then don't capture and return stderr, it's just confusing
    outerr, _ = p.communicate()
    return p.returncode, outerr